### PR TITLE
Rea-dd conv_affine_channel fuse pass as oneDNN only pass

### PR DESCRIPTION
**Pull request by piotrekobi**
### PR types

Performance optimization 
### PR changes

Others
### Describe

Readd pass removed in #39817 as a pass that only works with oneDNN enabled. This should fix the drop of the retinanet_r50_fpn_1x model reported in #40269. 
The model's latency decreases from 5289 to 3679, which means that the execution time is 30% faster. Also, the model's memory usage dropped by 62%, from 5701968KB to 2148480KB.

### CODE REVIEW

**Comment by piotrekobi** [comment]
@zhouwei25 please help with parallel_UT_rule.py approval

**Comment by piotrekobi** [comment]
@XieYunshen please help with TIMEOUT approval

**Comment by piotrekobi** [comment]
@lidanqing-intel @jczaja Please review

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
Hi, @jakpiase @wozna Do you know this this pass test has to be add into tool/parallel_UT_rule.py?
```
2022-05-05 18:21:33 2022-05-05 18:21:33 (2.58 MB/s) - 已保存 “bk.txt” [5/5])
2022-05-05 18:21:39 ****************
2022-05-05 18:21:40 0. You must have one RD (zhouwei25 (Recommend), wanghuancoder, luotao1) approval for tools/parallel_UT_rule.py changes, which manages the rule of running unittest with a same GPU. If the unittest failed due to Insufficient GPU memory or CUBLAS_STATUS_ALLOC_FAILED, you can remove it from tools/parallel_UT_rule.py.
2022-05-05 18:21:40 1. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting TIMEOUT properties.
```

File: python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt [hunk 1]
```diff
@@ -143,5 +143,6 @@ if (WITH_MKLDNN)
   set_tests_properties(test_mkldnn_conv_mish_fuse_pass PROPERTIES TIMEOUT 300)
   set_tests_properties(test_mkldnn_fc_mish_fuse_pass PROPERTIES TIMEOUT 300)
   set_tests_properties(test_mkldnn_fc_elementwise_add_fuse_pass PROPERTIES TIMEOUT 120)
+  set_tests_properties(test_mkldnn_conv_affine_channel_fuse_pass PROPERTIES TIMEOUT 60)
```
Comment (lidanqing-vv) [inline python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt:146]: I think this set_tests_properties is not necessary. If you remove this line, will all CIs pass too ?
Cause adding this line require Baidu's approval

File: tools/parallel_UT_rule.py [hunk 2]
```diff
@@ -1873,6 +1874,7 @@
     'test_dataloader_unkeep_order',
     'test_parallel_executor_profiler',
     'test_correlation',
+    'test_mkldnn_conv_affine_channel_fuse_pass',
```
Comment (lidanqing-vv) [inline tools/parallel_UT_rule.py]: I don't see mkldnn pass tests here. And it looks for tests running on GPU. Please confirm if adding mkldnn tests here is necessary

File: python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt (hunk 1 above)
Comment (piotrekobi) [reply python/paddle/fluid/tests/unittests/ir/inference/CMakeLists.txt:146]: It is necessary. Without it, there is a timeout error in the Coverage CI.

File: tools/parallel_UT_rule.py (hunk 2 above)
Comment (piotrekobi) [reply tools/parallel_UT_rule.py]: You're probably right. I left it because the old version added it here, but I'll try to remove it.

**Comment by piotrekobi** [comment]
@XieYunshen, @chalsliu Please help with TIMEOUT approval

**Comment by lidanqing-vv** [comment]
```
2022-05-09 16:12:22 2022-05-09 16:12:22 (3.50 MB/s) - 已保存 “bk.txt” [5/5])
2022-05-09 16:12:29 ****************
2022-05-09 16:12:29 0. You must have one QA (XieYunshen(Recommend) or chalsliu) approval for setting parameter RUN_TYPE as EXCLUSIVE, DIST, NIGHTLY, EXCLUSIVE:NIGHTLY or DISTNIGHTLY, or setting TIMEOUT properties.
```

**Review by lidanqing-vv — APPROVED** [review]
LGTM

**Review by jczaja — APPROVED** [review]
LGTM

**Review by XieYunshen — APPROVED** [review]
LGTM for `set_tests_properties(test_mkldnn_conv_affine_channel_fuse_pass PROPERTIES TIMEOUT 60)`

**Pull request merged**